src/riscv/lib/src/pvm/linux/signals.rs

```diff
@@ -166,50 +148,38 @@ impl Elem for LinuxSigAction {
 
 /// Information to support handling each supported signal
 pub struct SignalActions<M: ManagerBase> {
-    /// An array of [VirtAddr]s, unused
-    handlers: [Cell<VirtAddr, M>; SignalIndex::COUNT],
     /// An array of [VirtAddr]s, one action for each supported signal
     actions: [Cell<VirtAddr, M>; SignalIndex::COUNT],
-    /// An array of bitmasks, one mask for each supported signal
-    masks: [Cell<u32, M>; SignalIndex::COUNT],
     /// An array of bitmasks, one set of flags for each supported signal
     flags: [Cell<u32, M>; SignalIndex::COUNT],
-    /// An array of [VirtAddr]s, restorers for each signal, see
+    /// An array of bitmasks, one mask for each supported signal
+    masks: [Cell<u64, M>; SignalIndex::COUNT],
+    /// A pointer to a restorer address to jump to after returning from a signal handler. A
+    /// function to call `rt_sigreturn` should be loaded into here.
+    ///
     /// <https://www.man7.org/linux/man-pages/man2/sigreturn.2.html>
-    restorers: [Cell<VirtAddr, M>; SignalIndex::COUNT],
+    pub restorer: Cell<VirtAddr, M>,
```

Comment:
If you move the `write_restorer` function into this module, then you close the interface leakage. The field doesn't need to be public.